homeassistant/components/lyric/__init__.py

```diff
@@ -93,7 +95,16 @@ async def async_update_data() -> Lyric:
             async with async_timeout.timeout(60):
                 await lyric.get_locations()
             return lyric
-        except LYRIC_EXCEPTIONS as exception:
+        except LyricAuthenticationException as exception:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_REAUTH},
+                    data=entry.data,
+                )
+            )
+            raise UpdateFailed(exception) from exception
```

Comment:
My only idea here is to keep around some state that is checked before calling `get_locations`, though of course you'd have to remember to reset when the integration is reloaded.

Another alternative is to modify DataUpdateCoordinator to support a notion of a 'permanent failure', though I think it may be more straight forward to handle it within the integration until this becomes a widespread pattern.

Reply:
I did do a search and found the Mazda integration also has the same potential issue. It will be found on startup for now so shouldn't be too much of an issue